### DISCUSSION TITLE
Doc comment on Generative Search - OpenAI (/modules/reader-generator-modules/generative-openai)

### DISCUSSION THREAD

**Discussion by giscus** [weaviate/weaviate-io#806] [Docs/blog comments]
# Doc comment on Generative Search - OpenAI (/modules/reader-generator-modules/generative-openai)

In short

https://weaviate.io/developers/weaviate/modules/reader-generator-modules/generative-openai

**Comment by 1260684341** [comment]
how can i get the usage of openai？

**Reply by dandv** [reply]
Go to https://platform.openai.com/account/usage.

**Comment by edwineve** [comment]
Can't get it to work... i get normal results, but no generative search groupedresult in the response, it shows an OpenAI error:

"generate":{"error":"connection to: OpenAI API failed with status: 400 error: -536 is less than the minimum of 1 - 'max_tokens'","groupedResult":null}

query payload:

 {
 	"query": "{ Get {Article(limit: 10, nearText: {concepts: \"wagner crisis\"}) {title,body,_additional {id,certainty,generate(          groupedResult: {            task:            \"summarize the result\"                      }        ) {          groupedResult          error        }}}}}"
 }

Btw singleResult works fine!

**Comment by gepirvu** [comment]
I have the following spark code in Databricks to write into the Weaviate vector DB.
df.limit(100) 
    .write.format("io.weaviate.spark.Weaviate") 
    .option("scheme", "http") 
    .option("host", weaviate_network.split("https://")[1]) 
    .option("header:X-OpenAI-Api-Key", openai.api_key) 
    .option("className", class_name) 
    .mode("append") 
    .save()

But when I am running the code it is returning:
"Could not verify permissions for AppendData RelationV2[topic#351, link#352, domain#353, published_date#354, title#355, lang#356]  News, [scheme=http, host=stuttgart02-zf843mm1.weaviate.network, header:X-OpenAI-Api-Key" 

For the Weaviate cluster I am using public authentication. It's a poc. Wondering what kind of permission are expected here. Any suggestion would be great.

**Comment by jirawat-hirunkam-callminer** [comment]
The example properties in the generative-openai moduleConfig have wrong names. Please update the doc to use correct property names and also include new properties that have been added.

Correct names from https://github.com/weaviate/weaviate/blob/master/modules/generative-openai/config/class_settings.go :
"temperature"
"maxTokens"
"frequencyPenalty"
"presencePenalty"
"topP"
"baseURL"
"apiVersion"

Validation on bad property names should also be considered to prevent false assumptions that they're correct.
